### PR TITLE
[now-routing-utils] Fix segments in query string

### LINKED ISSUE #3540: now.json "rewrites" with named params have strange value once deployed
### Reduced test case

https://github.com/wilsonpage/zeit-now-next-rewrites-bug

### Steps:

1. Deploy with `now` (or use [this deploy](https://tmp-nextjs-example.wilsonpage.now.sh))
2. Visit https://tmp-nextjs-example.wilsonpage.now.sh/hello
3. Observe `world` `200` response ✅
4. Visit https://tmp-nextjs-example.wilsonpage.now.sh/hello-regex/john
5. Observe `{"name":"john","query":{"name":"john"}}` response ✅
6. Visit https://tmp-nextjs-example.wilsonpage.now.sh/hello-param/john
7. Observe `{"name":":name?name=john","query":{"name":":name?name=john"}}` response ❌

### PR DESCRIPTION
**Pull request by styfle**
This PR a regression when path segments are used in the query string.

Take a look at the following ASCII Table for why I had to delete certain parts of the parsed url before formatting again.

https://nodejs.org/api/url.html#url_url_strings_and_url_objects

Related to #3539
Fixes #3540

### CODE REVIEW

File: packages/now-routing-utils/test/superstatic.spec.js [hunk 1]
```diff
@@ -301,6 +301,15 @@ test('convertRewrites', () => {
     { source: '/some/old/path', destination: '/some/new/path' },
     { source: '/firebase/(.*)', destination: 'https://www.firebase.com' },
     { source: '/projects/:id/edit', destination: '/projects.html' },
+    {
+      source: '/users/:id',
+      destination: '/api/user?identifier=:id&version=v2',
+    },
```
Comment (ijjk) [inline packages/now-routing-utils/test/superstatic.spec.js:307]: should we also test a destination that has a segment in the `path` and `query` e.g. `/api/:user?identifier=:id`?

Comment (styfle) [reply packages/now-routing-utils/test/superstatic.spec.js:307]: @ijjk Great idea! Done ✅

Comment (ijjk) [reply packages/now-routing-utils/test/superstatic.spec.js:307]: Oh good catch with the hash one too 👍

**Review by ijjk — APPROVED** [review]
LGTM 👍

**Pull request merged**